doc/source/operations/ubuntu-jammy.rst

```diff
@@ -0,0 +1,523 @@
+
+   kayobe playbook run $KAYOBE_CONFIG_PATH/ansible/pulp-container-sync.yml
+   kayobe playbook run $KAYOBE_CONFIG_PATH/ansible/pulp-container-publish.yml
+
+Build locally customised container images
+-----------------------------------------
+
+.. note::
+
+   The container images provided by StackHPC Release Train are suitable for
+   most deployments. In this case, this step can be skipped.
+
+In some cases, it is necessary to build some or all images locally to apply
+customisations. To do this, set
+``stackhpc_pulp_sync_for_local_container_build`` to ``true`` before syncing
+container images.
+
+To build the overcloud images locally and push them to the local Pulp server:
+
+.. code-block:: console
+
+   kayobe overcloud container image build --push
+
+It is possible to build a specific set of images by supplying one or more
+image name regular expressions:
+
+.. code-block:: console
+
+   kayobe overcloud container image build --push ironic- nova-api
+
+Deploy the latest container images
+----------------------------------
+
+Make sure you deploy the latest containers before this upgrade:
+
+.. code-block:: console
+
+   kayobe seed service deploy
+   kayobe overcloud service deploy
+
+Common issues for all host types
+================================
+
+-  Interface names regularly change during upgrades, usually gaining the
+   ``np0`` suffix. This cannot easily be resolved. The upgrade script
+   configures networking both before and after rebooting to apply the upgrade.
+   Setting the interface statically in a kayobe-config fails during one of
+   these. This can be worked around by adding a ``sed`` command to the upgrade
+   script between the upgrade playbook step and the host configure step e.g.
+
+   .. code-block:: bash
+
+      kayobe playbook run $KAYOBE_CONFIG_PATH/ansible/ubuntu-upgrade.yml -e os_release=jammy --limit $1
+      sed -i -e 's/"ens1"/"ens1np0"/g' -e 's/"ens2"/"ens2np0"/g' $KAYOBE_CONFIG_PATH/environments/production/inventory/group_vars/compute/network-interfaces
+      kayobe overcloud host configure --limit $1 --kolla-limit $1 -e os_release=jammy
+
+   Remember to reset the change before upgrading another upgrade (or add a
+   second ``sed`` command to automate the process)
+-  Disk names can change during upgrades. This can be resolved in kayobe-config
+   once the new name is known (i.e. after the first upgrade) and applied by
+   re-running ``host configure`` for the affected host.
+-  Timeouts can become an issue with some hardware. The host will reboot once
+   or twice depending on whether it needs to apply package updates. Edit the
+   timeouts in the upgrade playbook (``ubuntu-upgrade.yml``) where required.
+
+Controllers
+===========
+
+Upgrade controllers *one by one*, ideally upgrading the host with the Virtual
+IP (VIP) last. Before upgrading a host with the VIP, restart the ``keepalived``
+container to fail it over to another controller.
+
+.. code-block:: bash
+
+   sudo docker restart keepalived
```

Comment:
I've found that just a restart often isn't a long enough downtime for the VIP to fail over. Usually I've needed to stop the container, wait a few seconds, then start it again.

Reply:
Done